packages/storybook-builder-vite/source-loader-plugin.js

```diff
@@ -1,24 +1,18 @@
-const inject =
-    require('@storybook/source-loader/dist/cjs/abstract-syntax-tree/inject-decorator').default;
+const transform =
+    require('@storybook/source-loader').default;
 
 module.exports.sourceLoaderPlugin = function () {
     return {
         name: 'storybook-vite-source-loader-plugin',
         enforce: 'pre',
-        transform(src, id) {
+        async transform(src, id) {
             if (id.match(/\.stories\.[jt]sx?$/)) {
-                const { source, sourceJson, addsMap } = inject(src, id);
-                const preamble = `
-                    /* eslint-disable */
-                    // @ts-nocheck
-                    // @ts-ignore
-                    var __STORY__ = ${sourceJson};
-                    // @ts-ignore
-                    var __LOCATIONS_MAP__ = ${JSON.stringify(addsMap)};
-                `;
+                // We need to mock 'this' when calling transform from @storybook/source-loader
+                const mockClassLoader = { emitWarning: () => {}, resourcePath: id };
```

Comment:
Could we maybe use `console.log` for `emitWarning`?  I'm not sure when/how it gets used, but it would be great not to trap/drop warnings if possible.

Reply:
Sure, it's only used when a parser can't be determined https://github.com/storybookjs/storybook/blob/c67c22afa75728eb00d782e88b4d43645d3a3253/lib/source-loader/src/abstract-syntax-tree/inject-decorator.js#L30